mobile/app/(auth)/landing.tsx

```diff
@@ -3,48 +3,93 @@ import "react-native-polyfill-globals/auto";
+import SideMenuAccountList from "@gno/components/list/account/account-list";
+import Layout from "@gno/components/layout";
+import Ruller from "@gno/components/row/Ruller";
+import { Spacer } from "@gno/components/row";
+import { useAuth } from "context/auth";
+import ReenterPassword from "@gno/components/modal/reenter-password";
+import Text from "@gno/components/text";
 
 export default function Page() {
+  const [accounts, setAccounts] = useState<GnoAccount[]>([]);
+  const [loading, setLoading] = useState<string | undefined>(undefined);
+  const [reenterPassword, setReenterPassword] = useState<GnoAccount | undefined>(undefined);
+
+  const gno = useGno();
+  const navigation = useNavigation();
+  const { signIn } = useAuth();
+
+  useEffect(() => {
+    const unsubscribe = navigation.addListener("focus", async () => {
+      try {
+        const response = await gno.listKeyInfo();
+
+        console.log("response", response);
+
+        setAccounts(response);
+      } catch (error: unknown | Error) {
+        console.log(error);
+      }
+    });
+    return unsubscribe;
+  }, [navigation]);
+
+  const onChangeAccountHandler = async (value: GnoAccount) => {
+    try {
+      setLoading("Changing account...");
+      const response = await gno.selectAccount(value.name);
+      setLoading(undefined);
+      if (!response.hasPassword) {
+        setReenterPassword(value);
+        return;
+      }
+
+      signIn({ name: value.name, password: "", pubKey: value.pubKey, address: value.address });
+    } catch (error: unknown | Error) {
+      setLoading(error?.toString());
+      console.log(error);
+    }
+  };
+
+  const onCloseReenterPassword = async (sucess: boolean) => {
+    if (sucess && reenterPassword) {
+      signIn({ name: reenterPassword.name, password: "", pubKey: reenterPassword.pubKey, address: reenterPassword.address });
+    }
+    setReenterPassword(undefined);
+  };
+
   return (
-    <View style={styles.container}>
-      <View style={styles.main}>
-        <Text style={styles.title}>GnoSocial</Text>
-        <Text style={styles.subtitle}>Experimental social dApp on Gno.land</Text>
-        <View style={styles.create}>
-          <Button.Link label="Sign in" href="/sign-in" />
-        </View>
-      </View>
-    </View>
+    <>
+      <Layout.Container>
+        <Layout.Body>
+          {accounts && accounts.length > 0 && (
+            <>
+              <Text.Body>Please, select one of the existing accounts to start:</Text.Body>
+              <SideMenuAccountList accounts={accounts} changeAccount={onChangeAccountHandler} />
+              <Spacer />
+              <Ruller />
+              <Spacer />
+              <Text.Body>Or use one of these options:</Text.Body>
+            </>
+          )}
+
+          <Spacer />
+          <Button.Link title="Sign in" href="/sign-in" />
+          <Spacer />
+          <Button.Link title="Sign Up" href="/sign-up" variant="tertiary" />
```

Comment:
The "Sign Up" button actually imports an account. Maybe this button should say "Import Account".